build.js

```diff
@@ -0,0 +1,174 @@
+};
+
+const formatEntries = (entries) =>
+  !entries
+    ? ""
+    : JSON.stringify(
+        Object.fromEntries(
+          Object.entries(entries)?.map(([k, v]) => {
+            const isStringOrNum =
+              typeof v === "string" || typeof v === "number";
+            return isStringOrNum ? [k, v] : flattenObj(k, v);
+          })
+        )
+      );
+
+const parseFontWeight = (value) => {
+  if (typeof value !== "string") return value;
+  const val = value?.toLowerCase();
+  return val === "regular"
+    ? 400
+    : val === "medium"
+    ? 500
+    : val === "semi bold" || val === "semibold"
+    ? 600
+    : val === "bold"
+    ? 700
+    : val === "black"
+    ? 800
+    : val;
+};
+
+const parseLetterSpacing = (value) => {
+  if (typeof value === "string") {
+    const lastChar = value.slice(-1);
+    if (lastChar === "%") {
+      let newBase = value.slice(0, value.length - 1);
+      return newBase + "em";
+    }
+    return value;
+  }
+  return `${value}px`;
+};
+
+const parseLineHeight = (value) =>
+  typeof value === "string" ? value : `${value}px`;
+
+const parseTypography = (obj = {}) => {
+  const { fontWeight, lineHeight, letterSpacing } = obj;
+  return {
+    ...obj,
+    fontWeight: parseFontWeight(fontWeight),
+    lineHeight: parseLineHeight(lineHeight),
+    letterSpacing: parseLetterSpacing(letterSpacing),
+  };
+};
+
+StyleDictionary.registerTransform({
+  name: "shadows/css",
+  type: "value",
+  matcher: ({ type }) => type === "boxShadow",
+  transformer: ({ value }) => {
+    const values = Array.isArray(value) ? value : [value];
+    const finalValues = values
+      .map(
+        ({ color, type, x, y, blur, spread }) =>
+          `${
+            type === "innerShadow" ? "inset " : ""
+          }${x}px ${y}px ${blur}px ${spread}px ${color}`
+      )
+      .join(", ");
+    return finalValues;
+  },
+});
+
+StyleDictionary.registerTransform({
+  name: "lineHeight/px",
+  type: "value",
+  matcher: ({ type }) => type === "lineHeights",
+  transformer: ({ value }) => parseLineHeight(value),
+});
+
+StyleDictionary.registerTransform({
+  name: "letterSpacing/em",
+  type: "value",
+  matcher: ({ type }) => type === "letterSpacing",
+  transformer: ({ value }) => parseLetterSpacing(value),
+});
+
+StyleDictionary.registerTransform({
+  name: "fontWeight/lowerCaseOrNum",
+  type: "value",
+  matcher: ({ type }) => type === "fontWeights",
+  transformer: ({ value }) => parseFontWeight(value),
+});
+
+StyleDictionary.registerTransform({
+  name: "typography/nested",
+  type: "value",
+  matcher: ({ type }) => type === "typography",
+  transformer: ({ value }) => parseTypography(value),
+});
+
+StyleDictionary.registerTransformGroup({
+  name: "js/custom",
+  transforms: [
+    "attribute/cti",
+    "name/cti/pascal",
+    "size/rem",
+    "color/hex",
+    "shadows/css",
+    "lineHeight/px",
+    "letterSpacing/em",
+    "fontWeight/lowerCaseOrNum",
+    "typography/nested",
+  ],
```

Comment:
Are any of these specific to tw? Wondering if this should be `js/tailwind` and have another for `js/mui`

Reply:
I imagine most, if not all of these transforms will be applicable for MUI, as well, but I won't be sure till I dig into MUI's theme requirements, which is my next to-do.